RELEASE.md

```diff
@@ -0,0 +1,101 @@
+- Configure a GitHub token for release scripts to use.
+  - https://github.com/settings/tokens
+  - Select the following permissions: 
+    - `admin:org`
+    - `public_repo`
+    - `read:user`
+  - Ensure your token is available to scripts, e.g. `export GH_API_TOKEN={YOUR-TOKEN}` or similar.
+- Get WPORG plugin svn commit access - ask a team member.
+
+_Outcome_: __You are equipped to ship a release!__
+
+## Release process
+### Lead-up to release
+#### Ensure release development is complete
+- Release milestone should exist (!).
+- Ensure all issues/PRs intended for this release are merged, closed and assigned to release milestone.
+- All PRs should have changelog entry, or `skip-changelog` tag.
+- Check with team to confirm any outstanding or in progress work.
+
+Note: changelog should be formatted like this in PR description. Note the preceding `> ` - this is required by changelog script.
+
+```md
+### Changelog
+
+> bug: Fix bug in Safari and other Webkit browsers that was causing the All Products block to show 0 results when resetting the sort value.
+```
+
+_Outcome_: __Team is aware of release and in agreement about what fixes & features are included.__
+
+#### Ensure release branch includes all relevant fixes
+- Make release branch if needed (major/minor).
+  - For fix releases, the branch should already exist.
+  - Otherwise create branch: `release/X.X`.
+- Update your local checkout to tip of the release branch.
+- Cherry pick all PRs into the release branch:
+  - If branch is already labelled `status: cherry-picked 🍒` then continue to next PR.
+  - Ideally, use GitHub Hub to cherry pick the PR - `hub am -3 {http://URL-TO-PR}`. 
+  - If there are serious conflicts or extensive differences between `master` and release branch, you may need to take more care:
+    - Manually cherry pick individual commits using git - `git cherry-pick {COMMIT-HASH}`.
+    - Or in some cases, manually craft a new PR with appropriate changes, targeting release branch.
+  - Push the release branch to origin (so changes are in GitHub repo).
+  - Label the PR: `status: cherry-picked 🍒`.
+
+_Outcome_: __Release branch has all relevant changes merged & pushed.__
+
+### Prepare release
+#### Ensure release branch readme is up to date
+- Run changelog script `npm run changelog` to get changelog txt for readme. Changelog content will be output to screen by script.
+- Add changelog section for release, e.g. `= 2.5.11 - 2020-01-20 =`.
+- Copy-paste the changelog content into `readme.txt`.
+- Make any other changes to readme as needed - e.g. support versions changing, new blocks. 
+- Push readme changes to release branch on origin repo.
+
+_Outcome_: __`readme.txt` is up to date with changes for release.__
+
+#### Build zip & smoke test
+- Ensure you are on tip of release branch.
+- Update dependencies – `npm ci`.
+- Run a production build - `npm run build`.
+- Run package script to get a zip to test `npm run package-plugin`.
+- Smoke test built release zip:
```

Comment:
I think it's also important to test on an existing site with blocks that were created before this release. This ensures that there are no unexpected block invalidation errors when editing that pre-existing content.

Reply:
Yes! 💌 this. 

As I get more details on all the testing involved in a release, it's starting to look like a significant chunk of time. Maybe as a next step we can map out how this looks in terms of time spent and people involved.

Reply:
Love this – I've now placed the testing emphasis on testing the zip before github/wporg release. It makes sense that we do the most thorough testing early, and the testing after that is quicker – confidence check to confirm no build / upload problems, test cross section of functionality.